jdaviz/configs/cubeviz/plugins/viewers.py

```diff
@@ -136,45 +142,62 @@ def _get_marks_for_layers(self, layers):
 
     def _on_subset_delete(self, msg):
         # delete any ShadowSpatialSpectral mark for which either of the spectral or spatial marks
-        # no longer exists
-        spectral_marks = self._get_marks_for_layers(self._get_spectral_subset_layers())
-        spatial_marks = self._get_marks_for_layers(self._get_spatial_subset_layers())
+        # no longer exists by matching the uuid of the msg subset to the uuid of the subsets
+        # in ShadowSpatialSpectral
+        spectral_marks = self._get_marks_for_layers([layer for layer in self._get_spectral_subset_layers() # noqa
+                                                     if layer.layer.uuid == msg.subset.uuid])
+        spatial_marks = self._get_marks_for_layers([layer for layer in self._get_spatial_subset_layers() # noqa
+                                                    if layer.layer.uuid == msg.subset.uuid])
         self.figure.marks = [m for m in self.figure.marks
-                             if not (isinstance(m, ShadowSpatialSpectral) and
-                                     (m.spatial_spectrum_mark in spatial_marks or
-                                      m.spectral_subset_mark in spectral_marks))]
+                             if not (isinstance(m, ShadowSpatialSpectral)
+                                     and msg.subset.uuid in [m.spatial_uuid, m.spectral_uuid])]
 
     def _expected_subset_layer_default(self, layer_state):
         """
         This gets called whenever the layer of an expected new subset is added, we want to set the
         default for the linewidth depending on whether it is spatial or spectral, and handle
         creating any necessary marks for spatial-spectral subset intersections.
         """
+        # TODO: this does not seem to be called correctly/in the right order when plugin data
+        #  is added AFTER subsets
```

Comment:
the problem here seems to be that adding a new cube (which will then be collapsed) is not triggering the creation of new `ShadowSpatialSpectral` marks.  We'll need to add a callback on adding data to the spectral viewer.  Questions for future us: does `_expected_subset_layer_default` get called in this case (or is it calling at the specviz-level instead of cubeviz)?